src/bci_build/service.py

```diff
@@ -0,0 +1,41 @@
+"""This module includes an abstraction over source services in the Open Build
+Service.
+
+"""
+
+import xml.etree.ElementTree as ET
+from dataclasses import dataclass
+from dataclasses import field
+from typing import Literal
+
+
+@dataclass(kw_only=True, frozen=True)
+class Service:
+    """Representation of an arbitrary source service in the Open Build Service."""
+
+    #: service mode (i.e. when the service runs)
+    mode: Literal["buildtime"] = "buildtime"
+
+    #: name of this service
+    name: str
```

Comment:
can we move this to the beginning, e.g. before mode? I prefer to have non-optional dataclass members first